accord-core/src/main/java/accord/utils/async/AsyncChain.java

```diff
@@ -63,6 +63,13 @@ default AsyncChain<Void> accept(Consumer<? super V> action, Executor executor)
         }, executor);
     }
 
+    default AsyncChain<V> withExecutor(Executor e)
```

Comment:
Random musings.

The abstraction async chain has where maybe you specify an executor and maybe you don't seems somewhat of an anti-pattern. I do think Guava got one thing right with `ListenableFuture` which is that specifying where your callback will execute is never optional.

But it's also pretty much implicit that it defaults to a direct execution. Still it's something of a hint to have to manually specify "direct execution", but also people will do it automatically a lot not knowing which thread it might execute on.

Reply:
the common case is that your "chain" is just a sequence of callbacks on the same thread, so we only ever need to define the executor at the start of the chain rather than maintain it for each step in the chain like you do with `ListenableFuture`, but you are right, not thinking about executors can be a problem and was one of the biggest issues getting this patch to work... but if people just put `direct` you kinda shot yourself in the foot just to compile =)